src/main/java/org/jabref/gui/ClipBoardManager.java

```diff
@@ -29,35 +34,43 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class ClipBoardManager {
-    public static final DataFormat XML = new DataFormat("application/xml");
+// Singleton with enum
+public enum ClipBoardManager {
+
+    INSTANCE;
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ClipBoardManager.class);
 
-    private final Clipboard clipboard;
-    private final ImportFormatReader importFormatReader;
+    private final Clipboard clipboard = Clipboard.getSystemClipboard();
+    private final java.awt.datatransfer.Clipboard primary = Toolkit.getDefaultToolkit().getSystemSelection();
+    private final ImportFormatReader importFormatReader = Globals.IMPORT_FORMAT_READER;
 
-    public ClipBoardManager() {
-        this(Clipboard.getSystemClipboard(), Globals.IMPORT_FORMAT_READER);
+    ClipBoardManager() {
     }
 
-    public ClipBoardManager(Clipboard clipboard, ImportFormatReader importFormatReader) {
-        this.clipboard = clipboard;
-        this.importFormatReader = importFormatReader;
+    public void watchInput(String newValue) {
+        if (!newValue.isEmpty()) {
+            primary.setContents(new StringSelection(newValue), null);
+        }
     }
 
-    /**
-     * Puts content onto the clipboard.
-     */
-    public void setContent(ClipboardContent content) {
-        clipboard.setContent(content);
+    public void inputToPrimary(TextInputControl input) {
+        input.insertText(input.getCaretPosition(), getContentsPrimary());
     }
 
+    // For activating this functionality in an input, use:
```

Comment:
It's nice that you provide code examples and the Javadoc has the `{@code` to make this easier (and in a standardized way)